File: backend/noava/noava/Models/ContactMessage.cs
using noava.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace noava.Models
{
    public class ContactMessage
    {
        public int Id { get; set; }

        [Required]
        public ContactTitle Title { get; set; }

        [Required]
        [EmailAddress]
        [MaxLength(255)]
        public string SenderEmail { get; set; } = null!;

        [Required]
        public ContactSubject Subject { get; set; }

        [Required]
        [MaxLength(4000)]
        public string Description { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/models/ContactMessage.ts
export type ContactTitle = 'Mr' | 'Mrs' | 'Ms' | 'Dr' | 'Other';

export type ContactSubject = 
  | 'GeneralInquiry' 
  | 'Support' 
  | 'BugReport' 
  | 'Feedback' 
  | 'Complaint' 
  | 'Other';

export type ContactMessageStatus = 'Pending' | 'InProgress' | 'Resolved';

interface ContactMessage {
  id: number;
  title: ContactTitle;
  senderEmail: string;
  subject: ContactSubject;
  description: string;
  status: ContactMessageStatus;
  createdAt: string;
}

interface ContactMessageRequest {
  title: ContactTitle;
  senderEmail: string;
  subject: ContactSubject;
  description: string;
}

interface ContactMessageFilter {
  subject?: ContactSubject;
  status?: ContactMessageStatus;
}

export type { ContactMessage, ContactMessageRequest, ContactMessageFilter };